### ISSUE TITLE
How to parse markdown fetched from Github using `remark`/`rehype` rather than `marked`?

### ISSUE THREAD

**Issue by ManasMadrecha** [nuxt/content#917]
Hi, on the `docs` theme, for the "Releases" page, I found that releases are fetched from Github API. And its `body`, i.e. the markdown is parsed using `this.$markdown`, where $markdown is a plugin.

https://github.com/nuxt/content/blob/0b6a7be6f6287cde880b0d37c88e6461efa167ff/packages/theme-docs/src/store/index.js#L103-L109

**But this plugin uses `marked` library.**
https://github.com/nuxt/content/blob/0b6a7be6f6287cde880b0d37c88e6461efa167ff/packages/theme-docs/src/plugins/markdown.js#L1

But since the Content module already has such a nice markdown parser using unified's extensive plugins of remark, rehype, etc, then is there some special reason to use a different library?

### Need
1. Simply convert the markdown to AST in the format as needed by Content module, and pass this `body` to the `nuxt-content` component's `document` prop to display the texts.
2. No need for importing a new library

### Issue I am facing / Help needed

Using remark-parser, I was able to convert the markdown body like this:

```js
unified().use(remarkParse).parse(release.body);
```

This converts to the JSON / AST format. But here:
- the `children` have `type` as `paragraph`, `heading` etc. 
- whereas, the Content module needs children to have `type` as `element` and their `tag` as `p`, `h2`, etc. 

So, basically Content module needs HTML sort of version. So, I used this:

```js
unified().use(remarkParse).use(remark2rehype).use(rehypeParse).parse(release.body);
```

Using this:
 - the `type` gets changed to `element`, which is good. BUT
 - there is no key called `tag`. It is called `tagName`. This can be replaced by `Object.keys`. BUT
 - new elements called 'html' and 'body' are now present in AST, and **there are no more deeply nested children which should have been created from the markdown**. Entire Github's markdown is in single string. 

So, can anyone well-versed with remark / rehype kindly help on how to properly convert this?

**ISSUE: Using `remark2rehype` isn't doing anything. Is there some specific option that needs to be passed to it, so that it converts MDAST to HAST?**

**Comment by ManasMadrecha** [comment]
## SOLVED

Instead of importing remark, rehype, etc. and reinventing the wheel to create my own markdown parser that too for just one "releases" page , I just used the already great Nuxt Content module.

## Steps:

### 1. Fetch the releases inside a custom remark plugin

Let's call this plugin "releases.js" and save it in `plugins/remark` folder.

```js
const fetch = require('node-fetch')

module.exports = function () {
  return async (tree, { data }) => {
    if (data.slug === 'releases') {
      let releases = []; 
     // let release = {};
      await fetch(
        "https://api.github.com/repos/<owner>/<repo-name>/releases"
      )
        .then(res => res.json())
        .then(
          res =>
          (releases = res
            .filter(r => !r.draft)
            .map(release => {
              return {
                url: release.html_url,
                name: release.name,
                tagName: release.tag_name,
                createdAt: release.created_at,
                publishedAt: release.published_at,
                body: release.body
              };
            }))
        );

      // if (releases.length) {
      //   release = releases.filter(
      //     release => release.tagName === data.slug
      //   );
      // }
      // release = releases ? releases[0] : {};
     // data.$release = release

      data.$releases = releases
    }
    return tree
  }
}
```

Here, note that we are mapping the fetched releases into a specific format of object. What all keys you need is up to you. I find the ones listed above are enough, but if you want you can also display contributors. Then, we are saving all our releases into `$releases` array. This we can access later in any of our code.

### 2. Register this plugin in nuxt config

Register it in `content.markdown.remarkPlugins` array.

```
['~/plugins/remark/releases.js']
```

### 3. IMPORTANT: Convert the Github's markdown into the format required by Nuxt Content

Do this inside nuxt config's hooks.

```js
'content:file:beforeInsert': async (document, database) => {
        if (document.$releases) {
          document.$releases.map(async release => {
            release.body = {
              body: await database.markdown.generateBody(release.body),
            };
            return release
          })
        }
}
```

### 4. Create a `releases.md` file anywhere in your `content` folder.

**Note:** If you want to also write your own custom contents in addition to the ones already written on Github releases, you can also create a `release` folder, and create multiple releases md files in that folder, like `1.1.0.md`, etc. 

If you do that, then it makes sense that instead of putting all the fetched releases into a single huge `$releases`, you can filter the fetched releases and save it inside a specific release file's `$release` key. Such code have been commented above. 

### 5. Create a `releases.vue` page and fetch the `releases.md` file there and display the body

```html
<template>
  <div>
    <article>
      <h1 class="tw-my-3 md:tw-my-6 tw-text-4xl tw-text-center">Releases</h1>
      <div v-if="post">
        <div
          v-for="release in post.$releases"
          :key="release.url"
        >
          <h2>{{release.tagName}}</h2>
          <nuxt-content :document="release.body"></nuxt-content>
        </div>
      </div>
    </article>
  </div>
</template>

<script>
export default {
  data() {
    return {
      post: {}
    };
  },
  async fetch() {
    this.post = await this.$content("releases").fetch();
  },
};
</script>
```
